ldp/alg/callbacks.py

```diff
@@ -123,54 +122,64 @@ async def after_update(self) -> None:
 
 
 class StoreTrajectoriesCallback(Callback):
-    """Simple callback that stores train/eval trajectories in an in-memory list."""
-
-    def __init__(self):
-        self.train_trajectories = []
-        self.eval_trajectories = []
-
-    async def after_train_step(self, trajectories: Sequence[Trajectory]) -> None:
-        self.train_trajectories.extend(trajectories)
-
-    async def after_eval_step(self, trajectories: Sequence[Trajectory]) -> None:
-        self.eval_trajectories.extend(trajectories)
-
-
-class TrajectoryFileCallback(Callback):
     """Callback that writes trajectories to a file."""
 
-    def __init__(self, output_dir: os.PathLike | str):
-        self.output_dir = Path(output_dir)
-        self.output_dir.mkdir(parents=True, exist_ok=True)
+    @staticmethod
+    def default_serialize_env(env: Environment, transition: Transition) -> str | None:
+        if not transition.done:
+            return None
+        try:
+            frame = env.export_frame()
+        except NotImplementedError:
+            return None  # Allow for envs that didn't implement export_frame()
+        return frame.model_dump_json(exclude={"state"}, indent=2)
```

Comment:
Per in-person convo: must we drop state here?

Reply:
Yeah I just kept this around to be backwards compatible, if one wants to serialize state, one can specify a serialization function.

Tho I do agree it's undesirable enough that we should not `exclude={"state"}` by default. @whitead what do you think of not `exclude`ing `state`?

Reply:
Alright, I just implemented this